File: iosApp/iosApp/Pages/StopDetails/TripStatus.swift
//
//  TripStatus.swift
//  iosApp
//
//  Created by esimon on 11/26/24.
//  Copyright © 2024 MBTA. All rights reserved.
//

import Shared
import SwiftUI

struct TripStatus: View {
    let predictions: UpcomingFormat

    var body: some View {
        switch onEnum(of: predictions) {
        case let .some(trips):
            if let trip = trips.trips.first {
                UpcomingTripView(
                    prediction: .some(trip.format),
                    routeType: trip.routeType
                )
            } else {
                EmptyView()
            }
        case let .disruption(alert):
            UpcomingTripView(prediction: .disruption(.init(alert: alert.alert), iconName: alert.iconName))
        case let .noTrips(format):
            UpcomingTripView(prediction: .noTrips(format.noTripsFormat))
        case .loading:
            UpcomingTripView(prediction: .loading)
        }
    }
}

File: androidApp/src/main/java/com/mbta/tid/mbta_app/android/stopDetails/TripStatus.kt
package com.mbta.tid.mbta_app.android.stopDetails

import androidx.compose.runtime.Composable
import com.mbta.tid.mbta_app.android.component.UpcomingTripView
import com.mbta.tid.mbta_app.android.component.UpcomingTripViewState
import com.mbta.tid.mbta_app.android.util.FormattedAlert
import com.mbta.tid.mbta_app.model.UpcomingFormat

@Composable
fun TripStatus(predictions: UpcomingFormat) {
    when (predictions) {
        is UpcomingFormat.Some ->
            when (val trip = predictions.trips.firstOrNull()) {
                null -> {}
                else ->
                    UpcomingTripView(
                        UpcomingTripViewState.Some(trip.format),
                        routeType = trip.routeType,
                    )
            }
        is UpcomingFormat.Disruption ->
            UpcomingTripView(
                UpcomingTripViewState.Disruption(
                    FormattedAlert(predictions.alert),
                    iconName = predictions.iconName,
                )
            )
        is UpcomingFormat.NoTrips ->
            UpcomingTripView(UpcomingTripViewState.NoTrips(predictions.noTripsFormat))
        UpcomingFormat.Loading -> UpcomingTripView(UpcomingTripViewState.Loading)
    }
}
